src/mono/wasm/wasm.proj

```diff
@@ -84,12 +84,29 @@
           AfterTargets="Build"
           DependsOnTargets="BuildPInvokeTable;BundleTimeZones">
 
-    <Exec Command="make -C $(MonoProjectRoot)wasm all SHELL=/bin/bash BINDIR=$(ArtifactsBinDir) MONO_BIN_DIR=$(MonoArtifactsPath) OBJDIR=$(ArtifactsObjDir) NATIVE_BIN_DIR=$(NativeBinDir) CONFIG=$(Configuration) PINVOKE_TABLE=$(WasmPInvokeTablePath) ICU_LIBDIR=$(PkgMicrosoft_NETCore_Runtime_ICU_Transport)/runtimes/browser-wasm/native/lib ENABLE_ES6=$(WasmEnableES6)"
+    <PropertyGroup>
+      <_IcuLibdir>$(PkgMicrosoft_NETCore_Runtime_ICU_Transport)/runtimes/browser-wasm/native/lib</_IcuLibdir>
+    </PropertyGroup>
+
+    <Exec Command="make -C $(MonoProjectRoot)wasm all SHELL=/bin/bash BINDIR=$(ArtifactsBinDir) MONO_BIN_DIR=$(MonoArtifactsPath) OBJDIR=$(ArtifactsObjDir) NATIVE_BIN_DIR=$(NativeBinDir) CONFIG=$(Configuration) PINVOKE_TABLE=$(WasmPInvokeTablePath) ICU_LIBDIR=$(_IcuLibdir) ENABLE_ES6=$(WasmEnableES6)"
           IgnoreStandardErrorWarningFormat="true" />
 
     <Copy SourceFiles="$(NativeBinDir)dotnet.js;
                        $(NativeBinDir)dotnet.wasm;
-                       $(NativeBinDir)dotnet.timezones.blat"
+                       $(NativeBinDir)dotnet.timezones.blat;
+                       $(MonoProjectRoot)wasm\runtime\driver.c;
+                       $(MonoProjectRoot)wasm\runtime\pinvoke.c;
+                       $(MonoProjectRoot)wasm\runtime\corebindings.c;
+                       $(MonoProjectRoot)wasm\runtime\pinvoke.h;
+                       $(MonoProjectRoot)wasm\runtime\binding_support.js;
+                       $(MonoProjectRoot)wasm\runtime\dotnet_support.js;
+                       $(MonoProjectRoot)wasm\runtime\library_mono.js;
+                       $(LibrariesProjectRoot)Native\Unix\System.Native\pal_random.js;
```

Comment:
I don't understand why it was needed before

Reply:
Because these were linked by emscripten into dotnet.js.

Reply:
we could bundle them so the runtime pack only has one

Reply:
Are you saying these are `dotnet_aot.js` ?

Reply:
These are the source files used to compile dotnet.wasm/dotnet.js in the runtime pack, to recompile them, we need to add them to the runtime pack.

Reply:
That's the part I don't understand why do we have to recompile js files and at the same time include dotnet.js?

Reply:
The JS files need to be reassembled completely because the emscripten configuration may change based on the configuration paremeters, and additionally linked bitcode files. This variability does change the `dotnet.js` file layout.